src/main/java/org/jboss/tm/usertx/TransactionListener.java

```diff
@@ -0,0 +1,16 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.Transaction;
+
+/**
+ * A listener callback for transaction related events
+ */
+public interface TransactionListener {
+    /**
+     * The callback for notifying registered listeners of a transaction related event
+     *
+     * @param transaction the transaction that the caller is registering interest in
+     * @param transactionEvent indication of what kind of change has occurred
+     */
+    void stateChange(Transaction transaction, TransactionEvent transactionEvent);
```

Comment:
Yes but then we have introduced a new class into the namespace when a simple parameter to the callback was sufficient. Additionally it sort of mirrors the registration interface which takes a separate parameter for the transaction.

But I accept that if we do ever want to add more payload to the event then your approach is more extensible than the current interface so I will change it.